Sources/Common/Extensions/DateExtension.swift

```diff
@@ -126,4 +126,17 @@ public extension Date {
         addingTimeInterval(timeInterval)
     }
 
+    func isSameDay(_ otherDate: Date?) -> Bool {
+        guard let otherDate = otherDate else { return false }
+        return Calendar.current.isDate(self, inSameDayAs: otherDate)
+    }
+
+    func isLessThan48HoursAgo() -> Bool {
+        self > Date().addingTimeInterval(-48 * 60 * 60)
```

Comment:
Technically `isLessThan(daysAgo: 2)` could be used as a backing implementation here. But that's just a nitpick.